src/TestSuite/ElasticStateReset.php

```diff
@@ -0,0 +1,79 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         4.0.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\ElasticSearch\TestSuite;
+
+use Cake\Datasource\ConnectionManager;
+use Cake\ElasticSearch\Datasource\Connection;
+use Cake\ElasticSearch\IndexRegistry;
+use Cake\TestSuite\Fixture\FixtureLoader;
+use Cake\TestSuite\Fixture\StateResetStrategyInterface;
+use Elastica\Query\MatchAll;
+
+class ElasticStateReset implements StateResetStrategyInterface
```

Comment:
I don't love this class name and would appreciate any alternatives. I'm also not sure if this should be in a `TestSuite\Fixture` namespace to follow the same convention as in cake core.

Once we have the names in a good place I'll add some tests.